src/org/traccar/database/ExtendedObjectManager.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2017 Anton Tananaev (anton@traccar.org)
+ * Copyright 2017 Andrey Kunitsyn (andrey@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.database;
+
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.traccar.Context;
+import org.traccar.helper.Log;
+import org.traccar.model.Device;
+import org.traccar.model.Group;
+import org.traccar.model.BaseModel;
+
+public abstract class ExtendedObjectManager extends SimpleObjectManager {
+
+    private final Map<Long, Set<Long>> deviceItems = new ConcurrentHashMap<>();
+    private final Map<Long, Set<Long>> deviceItemsWithGroups = new ConcurrentHashMap<>();
+    private final Map<Long, Set<Long>> groupItems = new ConcurrentHashMap<>();
+
+    protected ExtendedObjectManager(DataManager dataManager,
```

Comment:
Minor issue, but I usually prefer to move all parameters on the next line if it doesn't fit. I think it makes it easier to read like this:
```java
protected ExtendedObjectManager(
        DataManager dataManager, Class<? extends BaseModel> baseClass) {
```

Not a big deal. Just a comment for future code so we make styling more consistent across the codebase.